Assets/MixedRealityToolkit.SDK/Experimental/ScrollingObjectCollection/ScrollingObjectCollection.cs

```diff
@@ -0,0 +1,2232 @@
+                        bool isColliderActive = false;
+                        foreach (Collider col in c)
+                        {
+                            if (col.enabled)
+                            {
+                                isColliderActive = true;
+                                break;
+                            }
+                        }
+                        if (isColliderActive)
+                        {
+                            //Fire the UnityEvent
+                            ClickEvent?.Invoke(initialFocusedObject);
+                        }
+                    }
+
+                    ResetState();
+
+                }
+                else if (isDragging && canScroll)
+                {
+
+                    if (scrollDirection == ScrollDirectionType.UpAndDown)
+                    {
+                        //Lock X, clamp Y
+                        workingScrollerPos.y = MathUtilities.SoftClamp(initialScrollerPos.y - handDelta.y, minY, maxY, 0.5f);
+                        workingScrollerPos.x = 0.0f;
+                    }
+                    else
+                    {
+                        //Lock Y, clamp X
+                        workingScrollerPos.x = MathUtilities.SoftClamp(initialScrollerPos.x - handDelta.x, minX, maxX, 0.5f);
+                        workingScrollerPos.y = 0.0f;
+                    }
+
+                    //Update the scrollContainer Position
+                    ApplyPosition(workingScrollerPos);
+
+                    CalculateVelocity();
+
+                    //Update the prev val for velocity
+                    lastPointerPos = currentPointerPos;
+                }
+            }
+            else if (animateScroller == null && nodeLengthCheck)// Prevent the Animation coroutine from being overridden
+            {
+                //We're not engaged, so handle any not touching behavior
+                HandleVelocityFalloff();
+
+                //Apply our position
+                ApplyPosition(workingScrollerPos);
+            }
+        }
+
+        private void LateUpdate()
+        {
+            //Hide the items not in view
+            HideItems();
+        }
+
+        private void OnDisable()
+        {
+            //Unregister global input events
+            if (CoreServices.InputSystem != null)
+            {
+                CoreServices.InputSystem.UnregisterHandler<IMixedRealityInputHandler>(this);
+                CoreServices.InputSystem.UnregisterHandler<IMixedRealityTouchHandler>(this);
+                CoreServices.InputSystem.UnregisterHandler<IMixedRealityPointerHandler>(this);
+                CoreServices.InputSystem.UnregisterHandler<IMixedRealitySourceStateHandler>(this);
+            }
+
+            if (useOnPreRender && cameraMethods != null)
+            {
+                CameraEventRouter cameraMethods = CameraCache.Main.gameObject.EnsureComponent<CameraEventRouter>();
```

Comment:
should clippingBox.useOnPreRender = false now?

Reply:
what if clipping box needs to use OnPreRender for items unrelated to the ScrollingObjectCollection?